packages/openid4vc/src/openid4vc-holder/OpenId4vcSiopHolderService.ts

```diff
@@ -109,6 +110,14 @@ export class OpenId4VcSiopHolderService {
           challenge: nonce,
           domain: clientId,
           presentationSubmissionLocation: DifPresentationExchangeSubmissionLocation.EXTERNAL,
+          openid4vp: {
+            clientId,
+            verifierGeneratedNonce: nonce,
```

Comment:
I think this should be extracted from the `challenge` property in the pex service itself. As it's now redundant, and can lead to inconsistencies.

Reply:
I did this intentionally. I don't think we should use a challenge property. openid4vp is very specific. To me it would be completely unclear why the verification of a device response will fail after changing a e.g. domain/challenge property which have no relation at all to Mdoc?

Reply:
What do you think?

Reply:
Hmm I think it's quite common to use one nonce and apply that in all places that need a nonce. That way you keep consistency. we use the presentation exchange nonce in the w3c vp signature creation, so i don't see why we can't use it in the mdoc presentation creation.

I rather reuse the existing property, as there's really no reason to use a different nonce for the mdoc presentation.

Reply:
because the nonce for a w3c jwt vp / sd-jwt-vc is also based on the challenge. We just call it challenge in our api, but all VPs use this as the nonce/challenge.